src/layoutBuilder.js

```diff
@@ -388,75 +495,224 @@ function convertWordsToCodepoints(lineAsArrayOfCodepoints) {
+		// interacts with whitespace and punctuation in some cases)
+		let oldInlineIndex = inlinesBeforeTransformation.find(
+			(oldInline) => inline.text.trim() === oldInline.text.trim()
+		);
+		if (oldInlineIndex === -1) {
+			oldInlineIndex = inlinesBeforeTransformation.find((oldInline) =>
+				inline.text.includes(oldInline.text.trim())
+			);
+		}
+		if (oldInlineIndex === -1) {
+			oldInlineIndex = inlinesBeforeTransformation.find((oldInline) =>
+				oldInline.text.includes(inline.text.trim())
+			);
+		}
+		var oldInline = inlinesBeforeTransformation[oldInlineIndex];
+		var newInline = inline;
+		if (oldInline) {
+			newInline.style = oldInline.style;
+			newInline.background = oldInline.background;
+			newInline.font = oldInline.font;
+			newInline.decoration = oldInline.decoration;
+			newInline.decorationColor = oldInline.decorationColor;
+		}
+
+		return line.addInline(newInline);
+	});
+}
+
+function transformRtlInlines(inlines, node, styleStack, textTools) {
+	const wordPropsLookup = {};
+
+	// make a lookup for each of the words in the line, where we can keep their fonts and styles
+	// so that we can reapply them later
+	inlines.forEach((inline) => {
+		// in most cases this first case will suffice, but punctuation isn't handled consistently
+		// and so if we have a space inside the trimmed text, we'll split the lineand add those 
+		// strings as keys too to ensure a successful lookup
+		const text = inline.text.trim();
+		const lookupObject = {
+			font: 'NotoSansRTL',
+			style: inline.style,
+			alignment: inline.alignment,
+		};
+		wordPropsLookup[text] = lookupObject;
+		if (/\s/.test(text)) {
+			const keys = text.split(/\s/);
+			keys.forEach((key) => {
+				wordPropsLookup[key] = lookupObject;
+			});
+		}
+	});
+
+	// turn inlines into string
+	const stringOfInlines = inlines.map((i) => i.text).join('');
+
+	// make a bidi instance
+	const bidiString = bidi
+		.from_string(stringOfInlines, { direction: 'LTR' })
+		.reorder_visually();
+
+	const wordsAsArraysOfCodePoints = convertWordsToCodepoints(
+		bidiString.string_arr
+	);
+
+	const arrayOfTransformedWords = [];
+
+	wordsAsArraysOfCodePoints.forEach((wordAsCodePoints) => {
+		const wordAsString = spreadify(
+			String.fromCharCode,
+			String
+		)(wordAsCodePoints);
+		const bidiWord = bidi
+			.from_string(wordAsString, { direction: 'RTL' })
+			.reorder_visually()
+			.toString();
+
+		if (bidiWord !== ' ') {
+			let originalTextnode = wordPropsLookup[bidiWord.trim()] || {};
+			const newTextNode = Object.assign({}, originalTextnode, {
+				text: bidiWord,
+			});
+			arrayOfTransformedWords.push(newTextNode);
+		}
+	});
+
+	// need to add a space manually to the end of an RTL inline run
+	arrayOfTransformedWords[arrayOfTransformedWords.length - 1].text += ' ';
+
+	const updatedInlines = textTools.buildInlines(
+		[{ text: arrayOfTransformedWords, style: node.style }],
+		styleStack
+	);
+
+	return updatedInlines.items;
+}
```

Comment:
New function. This, broadly speaking, replicates the functionality of the previous `transformLineForRtl` function between lines 498 and 552.

The key difference is it makes a lookup to allow the styles to be found and reapplied to the individual words before the call to build the new inlines on line 658. This is what allows us to reattach the font/styling information (not possible in `transformLineForRtl`.

This has been separated out and is _only_ called for RTL terms - meaning it can reuse the existing `convertWordsToCodepoints` function to ensure parity with the `transformLineForRtl` function.